src/components/S/INumber.vue

```diff
@@ -1,23 +1,28 @@
 <script setup>
-import { ref, watch } from 'vue';
+import { computed } from 'vue';
 
 const props = defineProps({
-  modelValue: Number,
+  value: [Number, String],
   definition: Object
 });
 
-const emit = defineEmits(['update:modelValue']);
+const emit = defineEmits(['input']);
 
-const input = ref(props.modelValue || props.definition.default || undefined);
```

Comment:
Any reason to change this file? This code is used on others place we should keep ability to define default in `props.definition.default`

Reply:
I changed it because:
> update INumber component as there is no way to make it update from the outside (modalValue) is just the default value and turned it in more standard two-way bound input that accepts value and @input. This makes it compatible with v-model (which is not used here as we need to control value updates).

But now I see that `IString` works the same way, but it works somehow, I guess I need to learn more about Vue to figure out how it works (and why it doesn't work there).

Reply:
Ops sorry missed your explanations, I like your way better if it works and if we don't lose the ability to default to `props.definition.default`. Your way could be used in others components in S/ folder too. These components in S/ folder are made to be reusable on others repo to generate form based on JSONSchema.